src/analysis/normed_space/operator_norm.lean

```diff
@@ -1,193 +1,183 @@
-Authors: Jan-David Salchow, Sébastien Gouëzel
+Authors: Jan-David Salchow, Sébastien Gouëzel, Jean Lo
 
 The space of bounded linear maps
 
 Define the set of bounded linear maps between normed spaces and show basic facts about it. In
 particular
 
-(*) define a set L(E,F) of bounded linear maps between k normed spaces,
-(*) show that L(E,F) is a vector subspace of E → F,
-(*) define the 'operator norm' on L(E,F) and show that it induces the structure of a normed space
-    on L(E,F).
+(*) show that bounded linear maps are a vector subspace of E → F,
+(*) define the operator norm and show that it induces the structure of a normed space
+    on bounded linear maps.
 -/
 import algebra.module
 import analysis.normed_space.bounded_linear_maps
+import topology.metric_space.lipschitz
 
-variables {k : Type*}
-variables {E : Type*} {F : Type*}
+variables (k : Type*) (E : Type*) (F : Type*) (G : Type*)
 
+variables [normed_field k]
+variables [normed_space k E] [normed_space k F] [normed_space k G]
 
--- Define the subspace of bounded linear maps.
-section bounded_linear_maps
+noncomputable theory
 set_option class.instance_max_depth 50
 
+def bounded_linear_map : subspace k (E → F) :=
+{ carrier := {f : E → F | is_bounded_linear_map k f},
+  zero := is_bounded_linear_map.zero,
+  add := λ _ _, is_bounded_linear_map.add,
+  smul := λ _ _, is_bounded_linear_map.smul _ }
 
-variables [hnfk : normed_field k] [normed_space k E] [normed_space k F]
-include hnfk
+variables {k E F G}
 
-def bounded_linear_maps : subspace k (E → F) :=
-{ carrier := {A : E → F | is_bounded_linear_map k A},
-  zero := is_bounded_linear_map.zero,
-  add := assume A B, is_bounded_linear_map.add,
-  smul := assume c A, is_bounded_linear_map.smul c }
+namespace bounded_linear_map
 
--- introduce the local notation L(E,F) for the set of bounded linear maps.
-local notation `L(` E `,` F `)` := @bounded_linear_maps k E F _ _ _
+notation E ` →L[`:25 k `] ` F := bounded_linear_map k E F
 
 /-- Coerce bounded linear maps to functions. -/
-instance bounded_linear_maps.to_fun : has_coe_to_fun $ L(E,F) :=
+instance to_fun : has_coe_to_fun $ E →L[k] F :=
 {F :=  λ _, (E → F), coe := (λ f, f.val)}
 
-@[extensionality] theorem ext {A B : L(E,F)} (H : ∀ x, A x = B x) : A = B :=
-set_coe.ext $ funext H
-
-def to_linear_map (A : L(E, F)) : linear_map _ E F :=
-{to_fun := A.val, ..A.property}
+@[extensionality] theorem ext {f g : E →L[k] F} (h : ∀ x, f x = g x) : f = g :=
+set_coe.ext $ funext h
 
-@[simp] lemma bounded_linear_maps.map_zero {A : L(E, F)} : A (0 : E) = 0 :=
-(to_linear_map A).map_zero
+theorem ext_iff {f g : E →L[k] F} : f = g ↔ ∀ x, f x = g x :=
+⟨λ h x, by rintro; rw h, ext⟩
 
-@[simp] lemma bounded_linear_maps.coe_zero {x : E} : (0 : L(E, F)) x = 0 := rfl
+variables (c : k) (f g: E →L[k] F) (h : F →L[k] G) (x u v: E)
 
-@[simp] lemma bounded_linear_maps.smul_coe {A : L(E, F)} {x : E} {c : k} :
-  (c • A) x = c • (A x) := rfl
+def to_linear_map : linear_map _ E F :=
```

Comment:
Should there also be a coercion from bounded linear maps to linear maps, defined as `to_linear_map`?

Reply:
Added in a0d68f11f97256b340e23eada235975862c8350b .